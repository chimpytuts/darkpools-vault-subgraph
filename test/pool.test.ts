import { runTestCases } from './test-runner';
import { gql } from '@apollo/client/core';

const testCases = [
  {
    id: 'getPools',
    query: gql`
      query {
        pools {
          tokens {
            id
          }
          id
        }
      }
    `,
  },
];

runTestCases('Pools', testCases);
