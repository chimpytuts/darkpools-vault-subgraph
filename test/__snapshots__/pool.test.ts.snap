// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pools resolvers getPools 1`] = `
Object {
  "data": Object {
    "pools": Array [
      Object {
        "__typename": "Pool",
        "id": "0x52c1245a6e1948424a588a621605a9b88dd50e6f000200000000000000000003",
        "tokens": Array [
          Object {
            "__typename": "PoolToken",
            "id": "0x52c1245a6e1948424a588a621605a9b88dd50e6f000200000000000000000003-0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
          },
          Object {
            "__typename": "PoolToken",
            "id": "0x52c1245a6e1948424a588a621605a9b88dd50e6f000200000000000000000003-0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2",
          },
        ],
      },
      Object {
        "__typename": "Pool",
        "id": "0x75fe54ce4f6503c8d714d9497639608f591ac479000200000000000000000000",
        "tokens": Array [
          Object {
            "__typename": "PoolToken",
            "id": "0x75fe54ce4f6503c8d714d9497639608f591ac479000200000000000000000000-0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
          },
          Object {
            "__typename": "PoolToken",
            "id": "0x75fe54ce4f6503c8d714d9497639608f591ac479000200000000000000000000-0xcd80986f08d776ce41698c47f705cdc99ddbfb0a",
          },
        ],
      },
      Object {
        "__typename": "Pool",
        "id": "0x80cdbbe17a65107c8f14f38389dfb2fe036907e8000200000000000000000001",
        "tokens": Array [
          Object {
            "__typename": "PoolToken",
            "id": "0x80cdbbe17a65107c8f14f38389dfb2fe036907e8000200000000000000000001-0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
          },
          Object {
            "__typename": "PoolToken",
            "id": "0x80cdbbe17a65107c8f14f38389dfb2fe036907e8000200000000000000000001-0xf702269193081364e355f862f2cfbfcdc5db738c",
          },
        ],
      },
      Object {
        "__typename": "Pool",
        "id": "0x81b802515b0e8a583fc934bdac4cff214baa09ea000200000000000000000004",
        "tokens": Array [
          Object {
            "__typename": "PoolToken",
            "id": "0x81b802515b0e8a583fc934bdac4cff214baa09ea000200000000000000000004-0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
          },
          Object {
            "__typename": "PoolToken",
            "id": "0x81b802515b0e8a583fc934bdac4cff214baa09ea000200000000000000000004-0x5c0e66606eabec1df45e2add26c5df8c0895a397",
          },
        ],
      },
      Object {
        "__typename": "Pool",
        "id": "0xa6128d6807bbbf8530e44ee0f458f5f1f5f9d80d000200000000000000000002",
        "tokens": Array [
          Object {
            "__typename": "PoolToken",
            "id": "0xa6128d6807bbbf8530e44ee0f458f5f1f5f9d80d000200000000000000000002-0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2",
          },
          Object {
            "__typename": "PoolToken",
            "id": "0xa6128d6807bbbf8530e44ee0f458f5f1f5f9d80d000200000000000000000002-0xbf92d6171938ebf1c6c14bf436471694f7d282de",
          },
        ],
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
}
`;
